src/Microsoft.PowerShell.Commands.Utility/commands/utility/select-object.cs

```diff
@@ -337,6 +337,11 @@ private void ProcessExpressionParameter()
             if (ExcludeProperty != null)
             {
                 _exclusionFilter = new MshExpressionFilter(ExcludeProperty);
+                if ((Property == null) || (Property.Length == 0))
+                {
```

Comment:
To avoid this breaking change, we will have to do something like the following:
```
When -ExcludeProperty is specified
    #1 if -Property is specified
        always respect -ExcludeProperty
    #2 else if -ExpandProperty is specified
        Ignore -ExcludeProperty
    #3 else
        set Property to "*" and respect -ExcludeProperty
```
But this will again raise confusion about how `-ExcludeProperty` actually works (when and when not it takes effect), and probably someone will see `#2` as a bug too. Therefore, I'm more inclined to accept always having `#3` when `-Property` is not specified, though it would introduce the breaking change discussed above. 
@lzybkr @adityapatwardhan @vors, what are your opinions on this?